src/core/shutdown.c

```diff
@@ -146,6 +148,23 @@ static int parse_argv(int argc, char *argv[]) {
 }
 
 static int switch_root_initramfs(void) {
+        struct statfs st;
+        int r;
+
+        r = statfs("/run", &st);
+        if (r < 0)
+                log_error_errno(errno, "Failed to determine mount flags for /run: %m");
+
+        /* If /run is mounted with NOEXEC, remount without it, as otherwise exec /shutdown post switch_root will fail */
+        if (r == 0 && (st.f_flags & ST_NOEXEC)) {
+                if (mac_smack_use())
+                        r = mount(NULL, "/run", NULL, MS_REMOUNT|MS_NOSUID|MS_NODEV|MS_STRICTATIME, "mode=755,smackfsroot=*");
+                else
+                        r = mount(NULL, "/run", NULL, MS_REMOUNT|MS_NOSUID|MS_NODEV|MS_STRICTATIME, "mode=755");
+                if (r < 0)
+                        return log_error_errno(errno, "Failed to remount /run without MS_NOEXEC: %m");
```

Comment:
I don't think this should error out. Just print a warning and continue.

Reply:
If we failed to remount, execv of /shutdown will fail as the almost next action. I guess we don't care, right..... ?

Reply:
Yeah. The remounting is just an additional step, essentially a work-around for it being mounted wrong.